### ISSUE TITLE
Relicense the birdnetlib project with a more permissive open source software license

### ISSUE THREAD

**Issue by joeweiss** [joeweiss/birdnetlib#73]
Related to #72, I'm proposing to relicense the project as Apache 2.0. This would encourage the inclusion and usage of `birdnetlib` in other projects without imposing copyleft conditions in the [GPL-3.0 license](https://github.com/joeweiss/birdnetlib/blob/main/LICENSE).

Note: This would not apply to the models that are included in this repo, as they are included for redistribution under BirdNET-Analyzer and BirdNET-Lite's [CC-NC-BY-SA 4.0](https://github.com/joeweiss/birdnetlib/tree/main/src/birdnetlib/models) license.

The plan is to change the license in the 0.9.0 release in August. I'll leave this issue open for discussion until then.

**Comment by Bengt** [comment]
Please note that the terms of the CC-NC-BY-SA 4.0 require you to deliver a copy of the license with the licensed artifact. In this case, you would need to package the CC license alongside the Apache license into your Python distribution.

**Comment by Bengt** [comment]
My understanding is that the implication of different licenses would be, that the Python distributions of the birdnetlib GitHub project would be licensed under terms of *both* licenses. My understanding is that as a user of the birdnetlib Python distribution, I would not be able to use its functionality at all for commercial purposes, since I got wheel under the terms of both licenses. So even though the Apache 2.0 license allows for commercial use, users of birdnetlib would be prohibited because of BirdNET-Analyzer's license.

**Comment by Bengt** [comment]
> Users indeed have a responsibility to abide by the licenses of a library's dependencies as they apply to their usage. I don't understand the confusion. Installing a library with pip will install dependencies (eg watchdog, pydub, matplotlib, requests, tensorflow, numpy), many of which are not licensed under the same license as the library.

When the package manager is run by (and therefore on behalf of) the user of birdnetlib, they of course they need to consider the (one) license of every dependency when building their application. birdnetlib itself but has *two* licenses, both of which need to be abided by. ... That would at least be unusual.

**Comment by Bengt** [comment]
Yes, I noticed. Sorry about the confusion, deleting my earlier comment has had. On second thought, I found it to not be helpful, and I was assuming that you would not see it at all.

**Comment by Bengt** [comment]
> I have no idea if it is possible to acquire the models, or a compatible model, under a different license as I have no need for a ~non-commercial~ commercial license, but `birdnetlib` (as GPLv3 or Apache 2.0) would be able to be used in a commercial setting **if** a commercially-compatible licensed model existed (now or in the future).

Please consider that in some jurisdictions, commercial is a very widely defined term. In Germany, commercial can mean "a lot" in some contexts. So I could imagine running BirdNET-Analyzer on a whole dataset, maybe repeatedly over a longer time span as in bioreserve monitoring, might be considered commercial, even if done for research, as a hobby or for fun.

**Comment by joeweiss** [comment]
This is not the place for a discussion of BirdNET-Analyzer’s choice of license, or your personal interpretation of the meaning of its license.

Please limit the discussion to the issue at hand, which is the potential relicensing of this project. This is a small project, run by a single dev on a volunteer basis. Please stay on topic and if you have an issue or comment regarding the wider BirdNET community ecosystem, I suggest you take it somewhere else.

**Comment by joeweiss** [comment]
From reviewing commits and pull requests, it looks like @jurriaan and @LimitlessGreen have contributions in the current main branch. Thanks for your past contributions.

I've reached out to you both over email, but I'm also pinging you here. Please let me know if this license change (GPLv3 to Apache 2) is acceptable to you.

**Comment by jurriaan** [comment]
I'm fine with that! 👍

**Comment by LimitlessGreen** [comment]
> From reviewing commits and pull requests, it looks like @jurriaan and @LimitlessGreen have contributions in the current main branch. Thanks for your past contributions.
> 
> I've reached out to you both over email, but I'm also pinging you here. Please let me know if this license change (GPLv3 to Apache 2) is acceptable to you.

Hey,
sorry, hat somehow slipped past me 🙈
Sure, I'm completely fine with that! 👍
